src/app/external-cluster-wizard/steps/external-cluster/provider/aks/component.ts

```diff
@@ -306,9 +306,11 @@ export class AKSClusterSettingsComponent
           this.vmSizeLabel = this.vmSizes?.length ? VMSizeState.Ready : VMSizeState.Empty;
         });
 
-      this._getAKSKubernetesVersions();
-      this._getAKSLocations();
-      this._getAKSResourceGroups();
+      this._externalClusterService.presetChanges.pipe(startWith(null), takeUntil(this._unsubscribe)).subscribe(_ => {
```

Comment:
There are couple of scenarios:

Case-1: 
- Component gets destroyed when present is cleared from 2nd step via (x) icon -> Reinitialising make it work fine as new component gets initialised

Case-2:
- Component gets initialised on preset -> Navigate to 3rd step -> Navigate back 2nd step -> don't clear preset -> change present -> 3rd component is not destroyed BUT here the logic you have put place is **correct** BUT we will never get `null` until components gets destroyed EVEN if we get it it will be use less because it will be Case-1:

- check to run if value not null then trigger API calls  OR 
- Invoke API calls if preset value is coming in `_` parameter

Reply:
@Talha-Jamil-TJ why not use the same `BehaviorSubjet` approach that you used for region changes?

Reply:
@Waseem826 you are right, I'll change `presetChanges` to a behavior subject too.